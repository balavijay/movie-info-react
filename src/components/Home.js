import React from 'react';
import {
  POSTER_SIZE,
  IMAGE_BASE_URL,
} from '../config';

// import Components
import Grid from './elements/Grid/Grid';
import MovieThumb from './elements/MovieThumb/MovieThumb';

import Spinner from './elements/Spinner/Spinner';

// Custom Hook
import { useHomeFetch } from './hooks/useHomeFetch';

import NoImage from './images/no_image.jpg';

const Home = () => {
  const [
    {
      state: { movies  },
      loading,
      error,
    },

  ] = useHomeFetch();




  if (error) return <div>Something went wrong ...</div>;
  if (!movies[0]) return <Spinner />;

  return (
      <>
        <Grid header={'Landing Page'}>
          {movies.map(movie => (
              <MovieThumb
                  key={movie.id}
                  clickable
                  image={
                    movie.poster_path
                        ? IMAGE_BASE_URL + POSTER_SIZE + movie.poster_path
                        : NoImage
                  }
                  movieId={movie.id}
                  movieName={movie.original_title}
              />
          ))}
        </Grid>
        {loading && <Spinner />}

      </>
  );
};

export default Home;
