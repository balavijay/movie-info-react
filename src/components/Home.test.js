import React from 'react';
import { shallow, mount } from 'enzyme';

import Home from './Home';

describe ("Home Component " , () => {

   const  app = shallow(<Home />);

    test("Component rendered" , () => {
        expect(app).toMatchSnapshot();
    });

    test("Home component" , () => {

        const wrapper = mount(<Home   />);
        expect(wrapper.find('MovieThumb').length).toEqual(0);
    });



     
 
});