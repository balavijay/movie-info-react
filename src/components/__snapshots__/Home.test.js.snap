// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component  Component rendered 1`] = `ShallowWrapper {}`;
